evm/src/helpersV2.ts

```diff
@@ -35,6 +35,48 @@ interface RolesAndPersonas {
   personas: Personas
 }
 
+/**
+ * This helper function allows us to make use of ganache snapshots,
+ * which allows us to snapshot one state instance and revert back to it.
+ *
+ * This is used to memoize expensive setup calls typically found in beforeEach hooks when we
+ * need to setup our state with contract deployments before running assertions.
+ *
+ * @param provider The provider that's used within the tests
+ * @param cb The callback to execute that generates the state we want to snapshot
+ */
+export function useSnapshot(
+  provider: ethers.providers.JsonRpcProvider,
+  cb: () => Promise<void>,
+) {
+  const d = debug.extend('memoizeDeploy')
+  let hasDeployed = false
+  let snapshotId = ''
+
+  return async () => {
+    if (!hasDeployed) {
+      d('executing deployment..')
+      await cb()
+
+      d('snapshotting...')
+      /* eslint-disable-next-line require-atomic-updates */
+      snapshotId = await provider.send('evm_snapshot', undefined)
+      d('snapshot id:%s', snapshotId)
+
+      /* eslint-disable-next-line require-atomic-updates */
```

Comment:
Its because im accessing a variable that's been closure captured and is asynchronous too. But in our case this is safe since we don't run this function more than once at a time.